src/Analysers/HitFitAnalyser.cpp

```diff
@@ -29,144 +31,130 @@ BAT::TtbarHypothesis HitFitAnalyser::analyseAndReturn(const EventPtr event, cons
-	// TtbarHypothesisPointer bestTopHypothesis = chi2Reco->getBestSolution();
-	// cout << "Got results" << endl;
+	histMan_->setCurrentHistogramFolder(histogramFolder_);
 
 	//set MC matching flag
-	if (event->getDataType() == DataType::TTJets)
+	if (event->getDataType() == DataType::TTJets_amcatnloFXFX)
 		do_MC_matching = true;
 	else
 		do_MC_matching = false;
 
-	//initialise pile-up shifters
-	PoissonMeanShifter PShiftUp(0.6); // PU-systematic
-	PoissonMeanShifter PShiftDown(-0.6); // PU-systematic
-
-	//temporary file for jet fit info (ugly but works)
-	string tempFileName = "temp.txt";
-	ofstream tempFile(tempFileName.c_str());
-
-	if (Globals::produceFitterASCIIoutput) {
-		//write the event info into ASCII file
-		outFile << "------------------------------------------" << endl;
-		outFile << "BeginEvent" << endl;
-		outFile << "RunNr: " << event->runnumber() << endl;
-		outFile << "LumiSection: " << event->lumiblock() << endl;
-		outFile << "EventNr: " << event->eventnumber() << endl;
-		outFile << "EventWeight: " << weight_ << "  ";
-		double nVertices(0);
-		if (!event->isRealData()) {
-			float lumiWeight = event->PileUpWeight();
-
-			if (Globals::pileUpReweightingMethod == PileUpReweightingMethod::averagePileUp
-					|| Globals::pileUpReweightingMethod == PileUpReweightingMethod::threeDReweighting)
-				nVertices = event->averageNumberOfVertices();
-			if (Globals::pileUpReweightingMethod == PileUpReweightingMethod::inTimePileUpOnly)
-				nVertices = event->inTimeOnlyNumberOfVertices();
-			float lumiWeightUp = lumiWeight * PShiftUp.ShiftWeight(nVertices);
-			float lumiWeightDown = lumiWeight * PShiftDown.ShiftWeight(nVertices);
-			outFile << lumiWeight << "  " << lumiWeightUp << "  " << lumiWeightDown;
-
-		} else
-			outFile << "1  1  1";
-
-		outFile << endl;
-
-		outFile << jets.size() << "  " << event->Vertices().size() << "  ";
-		if (!event->isRealData()) {
-			outFile << nVertices;
-		} else
-			outFile << "9999";
-
-		outFile << "  ";
-		if ((event->getDataType() == DataType::TTJets) or (event->getDataType() == DataType::ttbar161)
-				or (event->getDataType() == DataType::ttbar163) or (event->getDataType() == DataType::ttbar166)
-				or (event->getDataType() == DataType::ttbar169) or (event->getDataType() == DataType::ttbar175)
-				or (event->getDataType() == DataType::ttbar178) or (event->getDataType() == DataType::ttbar181)
-				or (event->getDataType() == DataType::ttbar184)) {
-			outFile << "9999  9999  9999" << endl; //needs to be fixed
-		} else
-			outFile << "9999  9999  9999" << endl;
-	}
-
 	//prepare the jets collection
 	// Copy jets into an array
 	JetCollection jetCopy;
-	for (JetCollection::const_iterator j = jets.begin(); j != jets.end(); ++j) {
+	for (JetCollection::const_iterator j = leadingLightJets.begin(); j != leadingLightJets.end(); ++j) {
 		jetCopy.push_back(*j);
 	}
 
+	JetCollection bJetCopy;
+	for (JetCollection::const_iterator j = leadingBJets.begin(); j != leadingBJets.end(); ++j) {
```

Comment:
Initially I was thinking you copy jets here, but JetCollection is just a vector of Jet pointers.
All good.